src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -327,7 +331,10 @@ private async ValueTask<FlushResult> ProcessDataWrites()
                     }
                     else if (readResult.IsCompleted && _streamEnded)
                     {
-                        Debug.Assert(readResult.Buffer.Length == 0);
```

Comment:
This assertion never failed as far as I'm aware, but that was the only other Debug.Assert in the class, so I changed it so it also just results in a critical log if it fails.